src/iterators/bfs.rs

```diff
@@ -28,7 +29,7 @@ impl<'a, T> Bfs<'a, T> {
         Bfs {
             queue: VecDeque::with_capacity(graph.vertex_count()),
             current_ptr,
-            visited_stack: Vec::with_capacity(graph.vertex_count()),
+            visited_set: HashSet::new(),
```

Comment:
Maybe we can say here `HashSet::with_capacity(graph.vertex_count)` to avoid reallocation?

Reply:
Yeah. That's much better.